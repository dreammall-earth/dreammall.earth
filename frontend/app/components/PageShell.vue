<template>
  <v-app>
    <slot />
  </v-app>
</template>
