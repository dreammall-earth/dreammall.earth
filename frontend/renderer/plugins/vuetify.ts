// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { ThemeDefinition, createVuetify } from 'vuetify'
import { aliases as mdiAliases } from 'vuetify/iconsets/mdi'
import { VNumberInput } from 'vuetify/labs/VNumberInput'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

import { aliases as customAliases } from '#assets/icons'
import dark from '#assets/scss/dark.module.scss'
import light from '#assets/scss/light.module.scss'

function makeThemeFromCssModule(theme: CSSModuleClasses, isDark: boolean): ThemeDefinition {
  return {
    dark: isDark,
    colors: {
      background: theme.backgroundColor,
      surface: theme.surfaceColor,
      primary: theme.primaryColor,
      secondary: theme.secondaryColor,
      success: theme.successColor,
      info: theme.infoColor,
      warning: theme.warningColor,
      error: theme.errorColor,
      font: theme.fontColor,
      'dropdown-background': theme.dropdownBackgroundColor,
      'cockpit-highlight': theme.cockpitHighlightColor,
      'dm-panel-submit-color': theme.dmPanelSubmitColor,
      'dm-panel-submit-background-color': theme.dmPanelSubmitBackgroundColor,
      'dm-panel-text-input-color': theme.dmPanelTextInputColor,
      'dm-panel-call-action-button-color': theme.dmPanelCallActionButtonColor,
      'dm-panel-call-action-button-background-color': theme.dmPanelCallActionButtonBackgroundColor,
      'dm-panel-call-action-button-indicator-color': theme.dmPanelCallActionButtonIndicatorColor,
      'dm-panel-call-action-button-indicator-background-color':
        theme.dmPanelCallActionButtonIndicatorBackgroundColor,
      'dm-panel-reminder-text-color': theme.dmPanelReminderTextColor,
      'dm-panel-reminder-link-color': theme.dmPanelReminderLinkColor,
      'dm-switch-inactive-color': theme.dmSwitchInactiveColor,
      'dm-switch-active-color': theme.dmSwitchActiveColor,
    },
    variables: {
      'border-color': theme.borderColor,
      'border-opacity': theme.borderOpacity,
      // Vuetify colors don't support opacity, so we use variables instead.
      // Beware that HEX values won't work.
      'icon-background': theme.iconBackground,
      'bottom-menu-background': theme.bottomMenuBackground,
      'sidebar-background': theme.sidebarBackground,
      'cockpit-input-background': theme.cockpitInputBackground,
      'cockpit-element-background': theme.cockpitElementBackground,
      'cockpit-chip-background': theme.cockpitChipBackground,
      'cockpit-chip-background-2': theme.cockpitChipBackground2,
      'panel-background': theme.panelBackground,
      'panel-element-background': theme.panelElementBackground,
      'dm-panel-text-input-background-color': theme.dmPanelTextInputBackgroundColor,
      'dm-panel-background-color': theme.dmPanelBackgroundColor,
      'dm-panel-border-color': theme.dmPanelBorderColor,
      'dm-panel-reminder-text-background-color': theme.dmPanelReminderTextBackgroundColor,
      'dm-panel-reminder-link-background-color': theme.dmPanelReminderLinkBackgroundColor,
    },
  }
}

const themes = {
  light: makeThemeFromCssModule(light, false),
  dark: makeThemeFromCssModule(dark, true),
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    components: {
      VNumberInput,
    },
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
    icons: {
      aliases: { ...mdiAliases, ...customAliases },
    },
    theme: {
      defaultTheme: 'light',
      themes,
    },
  })
