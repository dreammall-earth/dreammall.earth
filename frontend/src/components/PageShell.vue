<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
