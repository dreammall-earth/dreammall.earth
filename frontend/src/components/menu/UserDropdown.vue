<template>
  <ul class="dropdown border-sm pa-3 pb-2 my-2 bg-dropdown-background">
    <li>
      <button class="sign-out" @click="signOut">
        <v-icon icon="$logout"></v-icon>{{ $t('buttons.signout') }}
      </button>
    </li>
  </ul>
</template>

<script lang="ts" setup>
import { inject } from 'vue'

import GlobalErrorHandler from '#plugins/globalErrorHandler'
import AuthService from '#src/services/AuthService'

const authService = inject<AuthService>('authService')

async function signOut() {
  try {
    await authService?.signOut()
  } catch (error) {
    GlobalErrorHandler.error('auth error', error)
  }
}
</script>

<style scoped lang="scss">
.dropdown {
  font-size: 14px;
  list-style: none;
  border-radius: 20px;
}

.dropdown li {
  padding: 5px;
}

.dropdown button {
  display: flex;
  align-items: center;
  width: 100%;
}

.dropdown button i {
  margin-top: -2px;
  margin-right: 8px;
}
</style>
