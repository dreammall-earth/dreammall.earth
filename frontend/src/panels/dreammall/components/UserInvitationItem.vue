<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
