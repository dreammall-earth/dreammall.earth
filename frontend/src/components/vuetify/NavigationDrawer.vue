<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
