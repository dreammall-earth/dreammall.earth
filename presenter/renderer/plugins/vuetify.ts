// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/lib/styles/main.sass'
import { useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
// eslint-disable-next-line import/no-namespace
import * as components from 'vuetify/lib/components/index.mjs'
// eslint-disable-next-line import/no-namespace
import * as directives from 'vuetify/lib/directives/index.mjs'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: any) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    components,
    directives,
  })
