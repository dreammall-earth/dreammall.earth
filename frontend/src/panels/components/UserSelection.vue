<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
