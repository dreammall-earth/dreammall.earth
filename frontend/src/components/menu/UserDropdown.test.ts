import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach, afterEach } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import UserDropdown from './UserDropdown.vue'

describe('UserDropdown', () => {
  const Wrapper = () => {
    return mount(VApp, {
      slots: {
        default: h(UserDropdown as Component),
      },
    })
  }

  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('click dark button', () => {
    beforeEach(async () => {
      await wrapper.find('button[data-test-theme-switch').trigger('click')
    })

    it('changes to dark theme', () => {
      expect(wrapper.vm.theme.global.name.value).toBe('dark')
    })

    describe('click botton again', () => {
      it('changes to light theme', () => {
        expect(wrapper.vm.theme.global.name.value).toBe('light')
      })
    })
  })
})
