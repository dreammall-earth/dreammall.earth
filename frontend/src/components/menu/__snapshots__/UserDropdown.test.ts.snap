// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDropdown > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    <ul
      class="dropdown border-sm pa-3 pb-2 my-2 bg-dropdown-background"
      data-v-67ab2386=""
    >
      <li
        data-v-67ab2386=""
      >
        <button
          data-test-theme-switch=""
          data-v-67ab2386=""
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate v-theme--light v-icon--size-default"
            data-v-67ab2386=""
          >
            <svg
              fill="none"
              height="22"
              viewBox="0 0 22 22"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 4V0H12V4H10ZM16.65 6.75L15.275 5.375L18.075 2.5L19.475 3.925L16.65 6.75ZM18 12V10H22V12H18ZM10 22V18H12V22H10ZM5.35 6.7L2.5 3.925L3.925 2.525L6.75 5.35L5.35 6.7ZM18.05 19.5L15.275 16.625L16.625 15.275L19.475 18.025L18.05 19.5ZM0 12V10H4V12H0ZM3.925 19.5L2.525 18.075L5.325 15.275L6.05 15.95L6.775 16.65L3.925 19.5ZM11 17C9.33333 17 7.91667 16.4167 6.75 15.25C5.58333 14.0833 5 12.6667 5 11C5 9.33333 5.58333 7.91667 6.75 6.75C7.91667 5.58333 9.33333 5 11 5C12.6667 5 14.0833 5.58333 15.25 6.75C16.4167 7.91667 17 9.33333 17 11C17 12.6667 16.4167 14.0833 15.25 15.25C14.0833 16.4167 12.6667 17 11 17ZM11 15C12.1 15 13.0417 14.6083 13.825 13.825C14.6083 13.0417 15 12.1 15 11C15 9.9 14.6083 8.95833 13.825 8.175C13.0417 7.39167 12.1 7 11 7C9.9 7 8.95833 7.39167 8.175 8.175C7.39167 8.95833 7 9.9 7 11C7 12.1 7.39167 13.0417 8.175 13.825C8.95833 14.6083 9.9 15 11 15Z"
                fill="currentColor"
              />
            </svg>
          </i>
           $t('menu.theme.dark')
        </button>
      </li>
      <li
        data-v-67ab2386=""
      >
        <button
          class="sign-out"
          data-v-67ab2386=""
        >
          <i
            aria-hidden="true"
            class="mdi mdi-logout mdi v-icon notranslate v-theme--light v-icon--size-default"
            data-v-67ab2386=""
          />
          $t('buttons.signout')
        </button>
      </li>
      <li
        data-v-67ab2386=""
      >
        <button
          data-v-67ab2386=""
        >
          <i
            aria-hidden="true"
            class="mdi mdi-email-multiple-outline mdi v-icon notranslate v-theme--light v-icon--size-default"
            data-v-67ab2386=""
          />
           $t('menu.invitationLinks')
        </button>
      </li>
    </ul>
    
  </div>
</div>
`;
